### PR TITLE
Feature/create post screen

### PR DESCRIPTION
**Pull request by abdallahFc**
Add CreatePost UI

This pull request adds the `CreatePost UI` to the project. The CreatePost UI allows users to create a new post and includes various UI components for input and interaction.

Changes Made:

- Added the `CreatePost`  XML layout file, which defines the UI elements and constraints for the `CreatePost` screen.
- Implemented the necessary styling for the UI components, including margins, font, text color, and text size.
- Added the `ValidateFieldUseCade`  to check the field.

Please review and provide feedback. Thank you!

image

### CODE REVIEW

File: app/src/main/res/layout/fragment_create_post.xml [hunk 1]
```diff
@@ -11,7 +12,121 @@
...
+            android:textColor="@color/color_text_primary"
+            android:textSize="@dimen/text_size_title"
+            app:layout_constraintStart_toEndOf="@id/imageViewAvatar"
+            app:layout_constraintTop_toTopOf="@id/imageViewAvatar"
+            tools:text="Username" />
+
+        <androidx.appcompat.widget.AppCompatButton
+            android:id="@+id/buttonState"
+            android:layout_width="wrap_content"
+            android:layout_height="24dp"
```
Comment (TarekIdrees) [inline app/src/main/res/layout/fragment_create_post.xml]: replace this with dimension values

File: app/src/main/res/layout/fragment_create_post.xml [hunk 2]
```diff
@@ -11,7 +12,121 @@
...
+            android:layout_width="@dimen/zero"
+            android:layout_height="wrap_content"
+            android:background="@drawable/create_post_camera_button"
+            android:gravity="start|center_vertical"
+            android:paddingHorizontal="@dimen/spacing_medium"
+            android:text="Camera"
+            app:backgroundTint="@color/cards"
+            app:icon="@drawable/ic_camera_square"
+            app:iconPadding="@dimen/spacing_normal"
+            app:iconTint="#4B8CED"
```
Comment (TarekIdrees) [inline app/src/main/res/layout/fragment_create_post.xml]: replace it with colors values

**Review by TarekIdrees — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_create_post.xml [hunk 3]
```diff
@@ -11,7 +12,145 @@
...
+                android:layout_height="match_parent"
+                android:layout_gravity="end"
+                android:layout_marginEnd="16dp"
+                android:clickable="true"
+                android:gravity="center"
+                android:text="Post"
+                android:onClick="@{() -> viewModel.createPost(editTextContent.getText().toString())}"
+                android:textColor="@color/primary"
+                android:fontFamily="@font/worksans_semibold"
+                android:paddingHorizontal="8dp"
```
Comment (msalarab) [inline app/src/main/res/layout/fragment_create_post.xml]: remove hard code

File: app/src/main/res/layout/fragment_create_post.xml [hunk 4]
```diff
@@ -11,7 +12,145 @@
...
+            android:id="@+id/imageViewAvatar"
+            android:layout_width="@dimen/profile_image_size"
+            android:layout_height="@dimen/profile_image_size"
+            android:layout_marginStart="@dimen/spacing_medium"
+            android:layout_marginTop="@dimen/spacing_medium"
+            android:padding="1dp"
+            app:layout_constraintStart_toStartOf="@id/viewBackground"
+            app:layout_constraintTop_toTopOf="@id/viewBackground"
+            app:shapeAppearanceOverlay="@style/ShapeAppearance.Material3.Corner.Full"
+            app:strokeColor="#E5E7E9"
```
Comment (msalarab) [inline app/src/main/res/layout/fragment_create_post.xml]: add to color file

File: app/src/main/res/layout/fragment_create_post.xml [hunk 5]
```diff
@@ -11,7 +12,145 @@
...
+            app:layout_constraintStart_toEndOf="@id/imageViewAvatar"
+            app:layout_constraintTop_toTopOf="@id/imageViewAvatar"
+            tools:text="Username" />
+
+        <androidx.appcompat.widget.AppCompatButton
+            android:id="@+id/buttonState"
+            android:layout_width="wrap_content"
+            android:layout_height="@dimen/spacing_large"
+            android:layout_marginTop="@dimen/spacing_small"
+            android:drawablePadding="@dimen/spacing_xsmall"
```
Comment (msalarab) [inline app/src/main/res/layout/fragment_create_post.xml]: 4dp or 8dp not 6

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -37,4 +37,16 @@
         <item name="android:textSize">@dimen/text_size_small</item>
     </style>
 
+    <style name="SharedButtonStyle" parent="Widget.MaterialComponents.Button.TextButton">
+        <item name="android:layout_marginTop">2dp</item>
```
Comment (msalarab) [inline app/src/main/res/values/styles.xml:42]: 2dp !

**Review by msalarab — CHANGES_REQUESTED** [review]
well job

**Review by AmeerAmjed — APPROVED** [review]

**Review by msalarab — APPROVED** [review]

**Pull request merged**